src/addons/Portal/Portal.js

```diff
@@ -182,6 +200,15 @@ class Portal extends Component {
   // Component Event Handlers
   // ----------------------------------------
 
+  handlePortalClick = (e) => {
+    if (!this.props.closeOnCloseClick) return
+    if (!domUtils.closest(e.target, '[data-close]')) return
```

Comment:
I think the main alternative to address the issues raised in the related issue are to make Modal an auto controlled component that handles "open" and have it implement
- closeIcon, and handle closing on click
- actions, which would be an array of buttons, with an "action" prop or something to determine if they are approve/deny/close buttons and handle them appropriately

What I like about the approach in this PR is:
- it's lightweight
- it stays consistent with to SUI-core (although they use classes, which are more brittle than data attributes imo)
- works regardless of how the user defines the underlying markup (children/shorthand)
- is usable outside of Modal, in any of our components that use portal (e.g. Adding a close button to a pop up) or in any user-defined components that use Portal directly